sensor/kubernetes/listener/watcher/crd/crdwatcher.go

```diff
@@ -51,7 +46,8 @@ type resourceEvent struct {
 }
 
 func (w *crdWatcher) startHandler() error {
-	handler := newCRDHandler(w.stopSig, w.resourceC)
+	eventC, handler := newCRDHandler(w.stopSig)
+	w.resourceC = eventC
```

Comment:
I can't comment in unchanged lines so commenting here.

In line `if !cache.WaitForCacheSync(w.stopSig.Done(), informer.HasSynced) {`, we could wait for the actual handler to sync instead of the informer (since we only have one handler in this informer). This might help with races and messages being lost, as `informer.HasSynced` does not guarantee the individual handlers have finished synching.